test/common/common.go

```diff
@@ -247,3 +248,31 @@ func IsAtLeastVersion(minVersion string) bool {
 	// Compare returns: 0 if ver1 == ver2, -1 if ver1 < ver2, or +1 if ver1 > ver2
 	return semver.Compare(ocpSemVer, minSemVer) >= 0
 }
+
+func CleanupHubNamespace(namespace string) func() {
+	return func() {
+		By("Deleting namespace " + namespace)
+
+		err := ClientHub.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+		if !k8serrors.IsNotFound(err) {
+			ExpectWithOffset(1, err).Should(BeNil())
+		}
+
+		// Wait for the namespace to be fully deleted before proceeding.
+		EventuallyWithOffset(1,
+			func() bool {
+				_, err := ClientHub.CoreV1().Namespaces().Get(
+					context.TODO(), namespace, metav1.GetOptions{},
+				)
+				isNotFound := k8serrors.IsNotFound(err)
+				if !isNotFound && err != nil {
+					GinkgoWriter.Printf("'%s' namespace 'get' error: %w", err)
+				}
+
+				return isNotFound
+			},
+			DefaultTimeoutSeconds*2,
```

Comment:
Since this has been a flake before, should we go ahead and make this longer?
```suggestion
			DefaultTimeoutSeconds*4,
```